### PR TITLE
Fixed error code checking for Win32 systems

### PR DESCRIPTION
**Pull request by nitro44x**
I came across your code here: https://www.reddit.com/r/cpp/comments/9c9kqz/do_cpp_users_wantneed_a_cross_platform_framework/e5ag74p

I've been stepping through it to learn about loading shared libraries and came across a possible error. FreeLibrary returns non-zero results on success, while dlclose() returns 0 on success. Also, it looks like the code always gets an error message and sets it, even if loading/unloading the shared library was successful.

### CODE REVIEW

File: src/SharedLibrary.cpp [hunk 1]
```diff
@@ -55,55 +55,63 @@ bool SharedLibrary::open(const char *filename)
...
-    LPTSTR msg = nullptr;
-    FormatMessage(
-       FORMAT_MESSAGE_FROM_SYSTEM |FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_IGNORE_INSERTS,
-       nullptr, GetLastError(), MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
-       (LPTSTR)&msg, 0, nullptr);
+    if (!mPriv->implementation) {
+        LPTSTR msg = nullptr;
+        FormatMessage(
+            FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_IGNORE_INSERTS,
+            nullptr, GetLastError(), MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
```
Comment (nitro44x) [inline src/SharedLibrary.cpp]: GetLastError() is returning "The operation completed successfully", so checking if msg != nullptr is not enough to check for errors.

**Comment by nitro44x** [comment]
I combined this PR with another since they are tiny

**Comment by traversaro** [comment]
cc @drdanz @diegoferigo did you saw this?

**Comment by drdanz** [comment]
> did you saw this?

Actually no, thanks for pinging me! :sweat_smile:

The patch looks good (thanks @nitro44x), I wonder if this is in conflict with @diegoferigo changes in the blockfactory repo.

**Comment by diegoferigo** [comment]
Nope, no conflicts. My changes are more on the factory side.

**Comment by drdanz** [comment]
Perfect, I'll go on and merge this

**Comment by drdanz** [comment]
Merged, thanks @nitro44x

**Comment by nitro44x** [comment]
No problem, glad to help

**Pull request merged**